[WIP] Enable more operators in Quantization-aware Training

zhenhuaw-me opened the pull request:
The first commit is moved to here from https://github.com/tensorflow/tensorflow/pull/31914 per @suharshs 's comments. The minor changes of the first commit serves as a initial, I'd like to have more discussions while this is in progress.

jackwish commented:
**About the *activations* and `Identity` operator**. I am curious that, why `Identity` was taken as a activation operators rather than passthrough in `contrib/quantize`, while as passthrough in TOCO?
https://github.com/tensorflow/tensorflow/blob/v2.0.0-rc0/tensorflow/contrib/quantize/python/quantize.py#L34-L35

Would you please help to give me a hint @suharshs and @raghuraman-k .

zhenhuaw-me committed:
Enable more operators in Quantization-aware Training

1. Abs, Sub and Concat are taken as source and intermediate
   operators of which the quantization representation of input
   and output could be different and insert FakeQuant operators
   there.
2. Tile, Squeeze, and ExpandDims are pass though operators which
   have same quantization representation of inputs and outputs.

jackwish commented:
This is an in-progress PR, which is not ready for review. However, there is [an announcement](https://groups.google.com/a/tensorflow.org/forum/#!topic/developers/IhU_w5nitKI) of deleting `contrib` from this TF repo at 9/18. My concern is that, how can we work on things like this after that, or what is the codebase?

Should we wait until the Keras related API later (I searched around, failed to find such API in this repo currently) per @suharshs's [feedback](https://groups.google.com/a/tensorflow.org/forum/#!searchin/addons/quantization-aware$20training%7Csort:date/addons/5Ve1EGAAwgQ/PML0k0BbCAAJ)? :
> contrib.quantize will be deprecated in 2.0, we are actively working on a 2.0 friendly Keras api to support quantization aware training.

jackwish commented:
As [Quantization-aware Training is moving to model-optimization toolkit](https://groups.google.com/a/tensorflow.org/d/msg/developers/dvxDWMjR8YQ/pQMGE51EBwAJ), this PR will not happen in the TF core repo. Closing now.
